app/experimenter/experiments/forms.py

```diff
@@ -433,20 +430,49 @@ def clean_population_percent(self):
     def is_valid(self):
         return super().is_valid() and self.variants_formset.is_valid()
 
+    @transaction.atomic
     def save(self, *args, **kwargs):
-        experiment = super().save(*args, **kwargs)
+        self.variants_formset.save()
+        return super().save(*args, **kwargs)
 
-        for form in self.variants_formset.forms:
-            if form.cleaned_data["DELETE"] and form.cleaned_data["id"]:
-                form.instance.delete()
-            elif form.is_valid():
-                form.instance.experiment = experiment
-                form.save(*args, **kwargs)
 
-        return experiment
+class ExperimentVariantsAddonForm(ExperimentVariantsBaseForm):
+
+    FORMSET_FORM_CLASS = ExperimentVariantAddonForm
+    FORMSET_CLASS = ExperimentVariantsFormSet
+
+    addon_name = forms.CharField(
+        required=False,
+        label="Add-on Name",
+        help_text=Experiment.ADDON_NAME_HELP_TEXT,
+    )
+    addon_experiment_id = forms.CharField(
+        required=False,
+        label="Active Experiment Name",
```

Comment:
Why is the label "Name"? and not "Identifier" or "ID"?

Reply:
Yeah I changed that to reflect the way it's represented in add-ons: https://github.com/mozilla/shield-studies-addon-utils/blob/master/test-addon/src/studySetup.js#L23

Reply:
And added some context to the help text: https://github.com/mozilla/experimenter/pull/1064/files#diff-00de22e6c754b3d410f1d3e631e1825eR378